src/oop/object_independence/medium/q5/MainTest.java

```diff
@@ -0,0 +1,197 @@
+                "Your " + className + " constructor does not correctly initialize the " + firstFieldName + " attribute.");
+        _assertEquals(value2, testClass.getFieldValue(classInstance, secondFieldName),
+                "Your " + className + " constructor does not correctly initialize the " + secondFieldName + " attribute.");
+    }
+
+    @Test
+    public void getTotalWinsIsDefinedCorrectly() {
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getSecondField, null), String.format(incorrectMethodDefinition, getSecondField, className));
+        assertTrue(testClass.hasModifier(getSecondField, null, "public"), String.format(incorrectModifierMessage, getSecondField, className));
+        assertTrue(testClass.hasReturnType(getSecondField, null, int.class), String.format(incorrectReturnType, getSecondField, className));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void teamClassHasCorrectGetTotalWinsMethod(String value1, int value2) throws Throwable {
+        Object[][] arguments = {
+                {value1, String.class},
+                {value2, int.class}
+        };
+        String[] getMethodModifiers = {"public"};
+        String incorrectGetMethodMessage = String.join(" ",
+                "Your", getSecondField, "method does not correctly get the value of the", secondFieldName, "attribute.");
+        String incorrectMethodDefinition = "The %s method in the %s class is not defined correctly. Make sure it is declared, spelt correctly, and has the correct parameters.";
+        String incorrectModifierMessage = "The %s method in the %s class has the wrong visibility modifier.";
+        String incorrectReturnType = "The %s method in the %s class has the incorrect return type.";
+        assertTrue(testClass.hasMethod(getSecondField, null), String.format(incorrectMethodDefinition, getSecondField, className));
+        assertTrue(testClass.hasModifier(getSecondField, null, "public"), String.format(incorrectModifierMessage, getSecondField, className));
+        assertTrue(testClass.hasReturnType(getSecondField, null, int.class), String.format(incorrectReturnType, getSecondField, className));
+        Object classInstance = testClass.createInstance(arguments);
+        Object getMethodOutput = testClass.callMethod(getSecondField, getMethodModifiers, classInstance);
+        _assertEquals(value2, getMethodOutput, incorrectGetMethodMessage);
+    }
+
+    private static Stream<Arguments> set1InputProvider() {
+        return Stream.of(
+                Arguments.of(5, 6),
+                Arguments.of(10, 20),
```

Comment:
Should also test if you pass the same value that it already is, as well as setting the value to 0 and setting the value from 0 to another number.